integration/pvtdata/data_purge_test.go

```diff
@@ -0,0 +1,143 @@
+
+		marblechaincodeutil.AddMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-0", "color":"blue", "size":35, "owner":"tom", "price":99}`, org2Peer0)
+	})
+
+	AfterEach(func() {
+		cancel()
+
+		if process != nil {
+			process.Signal(syscall.SIGTERM)
+			Eventually(process.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+		if network != nil {
+			network.Cleanup()
+		}
+		os.RemoveAll(testDir)
+	})
+
+	// 6.	The purge transaction takes effect only if the corresponding capability is set
+	//    - Add a few keys into a collection
+	//    - Issue a purge transaction without setting the new capability
+	//    - [Verify that the purge fails]
+	PIt("should fail with an error if the purge capability has not been enabled on the channel")
+
+	It("should prevent purged data being included in responses after the purge transaction has been committed", func() {
+		marblechaincodeutil.AssertPresentInCollectionM(network, channelID, chaincode.Name, `test-marble-0`, org2Peer0)
+		marblechaincodeutil.AssertPresentInCollectionMPD(network, channelID, chaincode.Name, "test-marble-0", org2Peer0)
+
+		marblechaincodeutil.PurgeMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-0"}`, org2Peer0)
+
+		marblechaincodeutil.AssertDoesNotExistInCollectionM(network, channelID, chaincode.Name, `test-marble-0`, org2Peer0)
+	})
+
+	// 1. User is able to submit a purge transaction that involves more than one keys
+	PIt("should accept multiple keys for purging in the same transaction")
+
+	// 2. The endorsement policy is evaluated correctly for a purge transaction under different endorsement policy settings (e.g., collection level/ key-hash based)
+	PIt("should correctly enforce collection level endorsement policies")
+	PIt("should correctly enforce key-hash based endorsement policies")
+	PIt("should correctly enforce other endorsement policies (TBC)")
+
+	// 3. Data is purged on an eligible peer
+	//    - Add a few keys into a collection
+	//    - Issue a purge transaction for some of the keys
+	//    - Verify that all the versions of the intended keys are purged while the remaining keys still exist
+	//    - Repeat above to purge all keys to test the corner case
+	PIt("should remove all purged data from an eligible peer")
+
+	// 4.	Data is purged on previously eligible but now ineligible peer
+	//    - Add a few keys into a collection
+	//    - Submit a collection config update to remove an org
+	//    - Issue a purge transaction to delete few keys
+	//    - The removed orgs peer should have purged the historical versions of intended key
+	PIt("should remove all purged data from a previously eligible peer")
+
+	// 5. A new peer able to reconcile from a purged peer
+	//    - Stop one of the peers of an eligible org
+	//    - Add a few keys into a collection
+	//    - Issue a purge transaction for some of the keys
+	//    - Start the stopped peer and the peer should reconcile the partial available data
```

Comment:
You can add one more test - any further writes to private data after the purge operation are not purged, as a side-effect of previous purge operation.